src/lib/cpd/CpdEngine.ts

```diff
@@ -7,38 +7,87 @@ import {Config} from '../util/Config';
 import {ENGINE, LANGUAGE, Severity} from '../../Constants';
 import * as engineUtils from '../util/CommonEngineUtils';
 import CpdWrapper from './CpdWrapper';
-import {FILE_EXTS_TO_LANGUAGE} from '../../Constants';
 import {uxEvents} from '../ScannerEvents';
 import crypto = require('crypto');
+import * as EnvVariable from '../util/EnvironmentVariable';
 
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'CpdEngine');
+const eventMessages = Messages.loadMessages("@salesforce/sfdx-scanner", "EventKeyTemplates");
+
+//  CPD supported languages: [apex, java, vf, xml]
+const FileExtToLanguage: Map<string, LANGUAGE> = new Map([
+	// apex
+	['cls', LANGUAGE.APEX],
+	['trigger', LANGUAGE.APEX],
+	// java
+	['java', LANGUAGE.JAVA],
+	// vf
+	['component', LANGUAGE.VISUALFORCE],
+	['page', LANGUAGE.VISUALFORCE],
+	// xml
+	['xml', LANGUAGE.XML],
+]);
+
+
+// exported for visibility in tests
+export const CpdRuleName = 'copy-paste-detected';
+export const CpdRuleDescription = 'Identify duplicate code blocks.';
+export const CpdRuleCategory = 'Copy/Paste Detected';
+export const CpdInfoUrl = 'https://pmd.github.io/latest/pmd_userdocs_cpd.html#refactoring-duplicates';
+export const CpdViolationSeverity = Severity.LOW;
+export const CpdLanguagesSupported: LANGUAGE[] = [...new Set (FileExtToLanguage.values())];
 
 export class CpdEngine extends AbstractRuleEngine {
 
-	public readonly ENGINE_ENUM: ENGINE = ENGINE.CPD;
-	public readonly ENGINE_NAME: string = ENGINE.CPD.valueOf();
+	private readonly ENGINE_ENUM: ENGINE = ENGINE.CPD;
+	private readonly ENGINE_NAME: string = ENGINE.CPD.valueOf();
+
 
 	private minimumTokens: number;
 	private logger: Logger;
 	private config: Config;
-	private initialized: boolean;
-	private cpdCatalog: Catalog;
-	private validCPDLanguages: LANGUAGE[];
-	
-
+	private initialized = false;
```

Comment:
Nitpick: There's no reason not to strongly-type this as a boolean: `private initialized: boolean = false;`

Reply:
That was my original code as well. Got a linting error:
```
/Users/rmohan/Development/sfdx-scanner/src/lib/cpd/CpdEngine.ts
  51:2  error  Type boolean trivially inferred from a boolean literal, remove type annotation  @typescript-eslint/no-inferrable-types
```

Reply:
Fair enough.